compiler/src/main/scala/org/scalajs/core/compiler/PrepJSInterop.scala

```diff
@@ -338,6 +339,16 @@ abstract class PrepJSInterop extends plugins.PluginComponent
             reporter.error(exp.pos, msg)
           }
         }
+
+        if (sym.isModule && (enclosingOwner is OwnerKind.JSNonNative)) {
+          def shouldBeExposed: Boolean = {
+            !sym.isSynthetic &&
+            !isPrivateMaybeWithin(sym)
+          }
+          if (shouldBeExposed)
+            sym.addAnnotation(ExposedJSMemberAnnot)
+        }
```

Comment:
Isn't this change unrelated to Scala.js defined objects? This will also expose normal objects (which is correct). I think this should be in a separate commit.